### PR TITLE
Add compare-klv applet

### PR DESCRIPTION
**Pull request by daniel-riehm**
This PR adds the `compare-klv` applet, which will pretty-print any differences found between the KLV contents of two files.

### CODE REVIEW

**Review by bradh — COMMENTED** [review]
If you sort, are you losing context in something like SDCC?

The case I'm thinking about is where the values are the same, but the previous entries (what ST 1010 calls the Refined Source List) are in a different order, so the SDCC doesn't have the same meaning.

**Comment by daniel-riehm** [comment]
> If you sort, are you losing context in something like SDCC?
> 
> The case I'm thinking about is where the values are the same, but the previous entries (what ST 1010 calls the Refined Source List) are in a different order, so the SDCC doesn't have the same meaning.

Good question. The `klv_1010_sdcc_flp` structure (`arrows/klv/klv_1010.h`) has the (in retrospect, not super well-named) `members` member, which contains the ordered tags of the Refined Source List. Two `klv_1010_sdcc_flp`s do not compare as equal unless their `members` are equal in addition to their actual data. So once an SDCC knows what should come before it, no important information is lost when reordering.

File: arrows/klv/applets/compare_klv.cxx [hunk 1]
```diff
@@ -0,0 +1,901 @@
...
+namespace kwiver {
+
+namespace arrows {
+
+namespace klv {
+
+namespace {
+
+// ----------------------------------------------------------------------------
+// Video or JSON file
```
Comment (chetnieter) [inline arrows/klv/applets/compare_klv.cxx:40]: At the very least I would suggest this interface and its derived classes be in a separate header and source file. I think we should consider moving the interface to an appropriate place in vital and putting the implementations in `arrows/core`. They are pretty light weight but I can see them being useful to other applications or tools.

**Review by chetnieter — CHANGES_REQUESTED** [review]

File: vital/algo/metadata_stream_from_map_io.h
Comment (chetnieter) [inline vital/algo/metadata_stream_from_map_io.h:1]: Empty file or a GitHub quirk that it is still listed?

**Review by chetnieter — APPROVED** [review]
Remove that empty file if it is really there.

File: vital/algo/metadata_stream_from_map_io.h
Comment (daniel-riehm) [reply vital/algo/metadata_stream_from_map_io.h:1]: Oops - I must have messed up a rebase somewhere.

**Pull request merged**